src/Storages/MergeTree/IMergeTreeDataPart.cpp

```diff
@@ -758,6 +758,16 @@ void IMergeTreeDataPart::loadColumns(bool require)
         column_name_to_position.emplace(column.name, pos++);
 }
 
+bool IMergeTreeDataPart::canParticipateInMerges(const StoragePolicyPtr & storage_policy) const
+{
+    /// `IMergeTreeDataPart::volume` describes space where current part belongs, and holds
+    /// `SingleDiskVolume` object which does not contain up-to-date settings of corresponding volume.
```

Comment:
Why not pass are_merges_allowed to the SingleDiskVolume in createVolumeFromReservation then? (like for VolumeRAID1)

Reply:
@azat Because it would be impossible to update that field during reload of configuration.

Reply:
I was asked about this because it is passed to RAID1 but not to the JBOD now, but since RAID1 is something that you cannot enable anyway I guess it is fine